packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts

```diff
@@ -303,6 +303,16 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
     `);
   }
 
+  if (isEs5SupportNeeded(projectRoot) &&
```

Comment:
This warning will trigger everywhere not only in `build`, but also `serve`, `test` etc..

Also I don't think this warning should be tight with differential loading because a user can opt not to have ES5 browser support and still encounter this issue.

For example when targeting `ES2017`, and non of the browsers in browserlist require ES5 but still they don't support ES2016+ features.

Note: we already display a message for when users target es2017 maybe we should improve that message?
https://github.com/angular/angular-cli/blob/5fc1f2499cbe57f9a95e4b0dfced130eb3a8046d/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L298

Reply:
We should improve the below check and use the enums
```ts
 if (wco.tsConfig.options.target === 4) { 
```

Reply:
>Also I don't think this warning should be tight with differential loading because a user can opt not to have ES5 browser support and still encounter this issue.

Not sure if this is a concern in such scenario. IMO, it's dangerous to think that if you have ES5 and ES2016 you cover the entire interval of ES5-ES2016 browsers continuously. If you have only ES2016 bundle, it's more obvious that you support only ES2016+ browsers.

Reply:
@alan-agius4 https://github.com/angular/angular-cli/pull/14213